thunder.install

```diff
@@ -435,3 +436,130 @@ function thunder_update_8106() {
 
   return $updateLogger->outputHtml();
 }
+
+/**
+ * Installs the thunder_riddle module.
+ *
+ * This replaces the riddle paragraphs by a new one, using media_entity as
+ * a base. After that you are able to select riddles out of the entity browser.
+ */
+function thunder_update_8107() {
+
+  $entityTypeManager = \Drupal::entityTypeManager();
+
+  $paragraphType = $entityTypeManager->getStorage('paragraphs_type')
+    ->load('paragraphs_riddle_marketplace');
+
+  $thunderUpdater = \Drupal::service('thunder_updater');
+
+  /** @var \Drupal\thunder_updater\UpdateLogger $updateLogger */
+  $updateLogger = \Drupal::service('thunder_updater.logger');
+  $updateLogger->cleanLogs();
+
+  $successful = TRUE;
+  if (!empty($paragraphType)) {
+    if (\Drupal::service('module_installer')->install(['thunder_riddle'])) {
+
+      // Import Riddles.
+      \Drupal::service('media_riddle_marketplace')->createMediaEntities();
+
+      /** @var \Drupal\paragraphs\Entity\Paragraph[] $paragraphs */
+      $paragraphs = $entityTypeManager->getStorage('paragraph')
+        ->loadByProperties([
+          'type' => 'paragraphs_riddle_marketplace',
+        ]);
+
+      foreach ($paragraphs as $oldParagraph) {
+
+        $fieldName = $oldParagraph->get('parent_field_name')->get(0)->getString();
+
+        $host = $oldParagraph->getParentEntity();
+        if ($host) {
+
+          foreach ($host->get($fieldName)->getValue() as $delta => $field) {
+            if ($field['target_id'] == $oldParagraph->id()) {
+
+              /* @var \Drupal\media_entity\Entity\Media */
+              $media = $entityTypeManager->getStorage('media')
+                ->loadByProperties([
+                  'bundle' => 'riddle',
+                  'field_riddle_id' => pathinfo(parse_url($oldParagraph->get('field_link')->uri)['path'])['filename'],
+                ]);
+
+              $media = current($media);
+
+              $paragraph = Paragraph::create([
+                'type' => 'riddle',
+                'field_riddle' => $media->id(),
+              ]);
+              $paragraph->save();
+
+              $host->get($fieldName)->set($delta, [
+                'target_id' => $paragraph->id(),
+                'target_revision_id' => $paragraph->getRevisionId(),
+              ]);
+
+              if ($host->save()) {
+                $updateLogger->info(t('Converted @riddle in @name.', ['@riddle' => $media->label(), '@name' => $host->label()]));
+                try {
+                  $oldParagraph->delete();
+                }
+                catch (EntityStorageException $storageException) {
+                  $updateLogger->warning(t('Could not delete @riddle in @name.', ['@riddle' => $media->label(), '@name' => $host->label()]));
```

Comment:
I think also in this case ```$successful``` should be marked as ```FALSE```. As long we have one paragraph not deleted, removing of type can be dangerous.